### PR TITLE
[Calling][Feature] call start time event

### PR DESCRIPTION
**Pull request by iaulakh**
## Purpose

* Event from native SDK for call start time

## Does this introduce a breaking change?

- [ ] Yes
- [ ] No

## Pull Request Type
What kind of change does this Pull Request introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Pull Request Checklist

- [ ] Public API changes
- [ ] Verified for cross-platform
- [ ] Synced with iOS, Web team
- [ ] Internal review completed
- [ ] UI Changes
  - [ ] Screen captures included
  - [ ] Tested for Light/Dark mode
  - [ ] Tested for screen rotation
  - [ ] Tested on Tablet and small screen device (5")
  - [ ] Include localization changes
- [ ] Tests
  - [ ] Memory leak analysis performed
  - [ ] Tested on API 21, 26 and latest 
  - [ ] Tested for background mode
  - [ ] Unit Tests Included
  - [ ] UI Automated Tests Included

## How to Test

* Open...
* Click on...

## What to Check
Verify that the following are valid
* ...

### CODE REVIEW

File: azure-communication-ui/calling/src/main/java/com/azure/android/communication/ui/calling/models/CallCompositeCallStartTimeEvent.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+ */
+public class CallCompositeCallStartTimeEvent {
+    private final Date startTime;
+
+    /**
+     * Create {@link CallCompositeCallStartTimeEvent} with call start time.
+     *
+     * @param startTime call start time.
+     */
+    CallCompositeCallStartTimeEvent(final Date startTime) {
```
Comment (pavelprystinka) [inline azure-communication-ui/calling/src/main/java/com/azure/android/communication/ui/calling/models/CallCompositeCallStartTimeEvent.java:19]: Do we anticipate more properties in the Event? ARB will ask why not to use `Date` : `CallCompositeEventHandler<Date>`

File: azure-communication-ui/calling/src/main/java/com/azure/android/communication/ui/calling/CallComposite.java [hunk 2]
```diff
@@ -607,6 +608,36 @@ public void removeOnRemoteParticipantLeftEventHandler(
             final CallCompositeEventHandler<CallCompositeRemoteParticipantLeftEvent> eventHandler) {
         configuration.getCallCompositeEventsHandler().removeOnRemoteParticipantLeftEventHandler(eventHandler);
     }
+
```
Comment (pavelprystinka) [inline azure-communication-ui/calling/src/main/java/com/azure/android/communication/ui/calling/CallComposite.java:611]: Should we expose `public Date getCallStartTime()`?

File: azure-communication-ui/calling/src/main/java/com/azure/android/communication/ui/calling/redux/middleware/handler/CallingMiddlewareActionHandler.kt [hunk 3]
```diff
@@ -1005,4 +1008,21 @@ internal class CallingMiddlewareActionHandlerImpl(
...
+    /*  <CALL_START_TIME> */
+    private fun subscribeToCallStartTimeUpdates(store: Store<ReduxState>) {
+        coroutineScope.launch {
+            callingService.getCallStartTimeSharedFlow().collect { date ->
+                store.dispatch(CallingAction.CallStartTimeUpdated(date))
+                configuration.callCompositeEventsHandler.getOnCallStartTimeUpdatedHandlers().forEach {
+                    try {
+                        it.handle(date)
+                    } catch (ex: Exception) {
+                        // catching and suppressing any client's exceptions
```
Comment (pavelprystinka) [inline azure-communication-ui/calling/src/main/java/com/azure/android/communication/ui/calling/redux/middleware/handler/CallingMiddlewareActionHandler.kt:1021]: Would be nice to log exception.

**Review by pavelprystinka — APPROVED** [review]

File: azure-communication-ui/calling/src/main/java/com/azure/android/communication/ui/calling/redux/middleware/handler/CallingMiddlewareActionHandler.kt (hunk 3 above)
Comment (iaulakh) [reply azure-communication-ui/calling/src/main/java/com/azure/android/communication/ui/calling/redux/middleware/handler/CallingMiddlewareActionHandler.kt:1021]: Thanks Pavel. Will do for GA

**Review by ShaunaSong — APPROVED** [review]

**Pull request merged**